### PR TITLE
Improvements to the JSON section

### PR DESCRIPTION
**Pull request by royka**
Thanks for merging my first PR @kennethreitz !
sorry for leaving my original PR hanging after it had been reviewed

I've done some more improvements to the section with help from mplewis (https://github.com/mplewis )

### CODE REVIEW

**Comment by royka** [comment]
As with before if anyone can review this I would be very grateful!

File: docs/scenarios/json.rst [hunk 1]
```diff
@@ -1,33 +1,39 @@
-JSON parsing
+JSON
 ===========
```
Comment (sigmavirus24) [inline docs/scenarios/json.rst]: The number of `=`s should match the length of the title.

File: docs/scenarios/json.rst [hunk 2]
```diff
@@ -1,33 +1,39 @@
...
 
-json
------
+The `json <https://docs.python.org/2/library/json.html>`_ library can read JSON strings into a Python dictionary or list. It can also serialize Python dictionaries or lists into JSON strings.
```
Comment (sigmavirus24) [inline docs/scenarios/json.rst]: ``rst
The `json <...>`_ library can parse JSON from strings or files. When parsing, the library converts the data into native python types. The ``json`` library can also emit JSON as a string (or write it to a file) from native python types as well.
```

File: docs/scenarios/json.rst [hunk 3]
```diff
@@ -1,33 +1,39 @@
...
 
-`json <https://docs.python.org/2/library/json.html>`_ is a standard libary which can convert JSON to a Dictionay.
+* There are six basic types in JSON: objects, lists, numbers, strings, booleans, and null.
```
Comment (sigmavirus24) [inline docs/scenarios/json.rst]: I'm not sure we should be describing JSON in detail.

File: docs/scenarios/json.rst [hunk 4]
```diff
@@ -1,33 +1,39 @@
...
+* The root element of JSON representation is an object, signified by ``{ ... }``. JSON objects are analogous to Python dictionaries: they have keys which correspond to values.
+* JSON does not use single quotes. JSON exclusively uses double quotes. Using single quotes in the place of double quotes is invalid JSON syntax.
 
-For example, a JSON string like this:
+Parsing JSON
+------------
```
Comment (sigmavirus24) [inline docs/scenarios/json.rst]: Add a new line after this.

File: docs/scenarios/json.rst [hunk 5]
```diff
@@ -1,33 +1,39 @@
...
+The `json <https://docs.python.org/2/library/json.html>`_ libary is imported like this:
```
Comment (sigmavirus24) [inline docs/scenarios/json.rst]: You don't need to link to the documentation more than once.

File: docs/scenarios/json.rst [hunk 6]
```diff
@@ -1,33 +1,39 @@
...
 
 .. code-block:: python
 
-    "{'first_name':'Guido','last_name':'Rossum'}"
+    import json
```
Comment (sigmavirus24) [inline docs/scenarios/json.rst]: We don't need a separate block to demonstrate how to import the library.

File: docs/scenarios/json.rst [hunk 7]
```diff
@@ -1,33 +1,39 @@
...
 
-can be loaded like this:
+Take the following string containing JSON data:
+
+.. code-block:: python
+
+    json_string = '{"first_name": "Guido", "last_name":"Rossum"}'
+
+It can be manpulated like this:
```
Comment (sigmavirus24) [inline docs/scenarios/json.rst]: You're parsing the string, not manipulating it.

File: docs/scenarios/json.rst [hunk 8]
```diff
@@ -1,33 +1,39 @@
...
+and can now be used as a normal dictionary:
 
 .. code-block:: python
 
     converted_dict['first_name']
 
-As well as converting a JSON string to a dictionary. You can convert a dictionary to JSON
-
-For example, given:
+As well as converting a JSON string to a dictionary. You can convert a dictionary to JSON:
```
Comment (sigmavirus24) [inline docs/scenarios/json.rst]: Please remove the sentence beginning with "As well as ... "

File: docs/scenarios/json.rst [hunk 9]
```diff
@@ -36,25 +42,24 @@ For example, given:
         'second_name': 'Rossum'
     }
 
-    import json
-    print json.dumps(d)
+    print(json.dumps(d))
     "{'first_name':'Guido','last_name':'Rossum'}"
 
-It is also possible to import JSON files:
+We can also load a JSON file by using ``json.load`` instead of ``json.loads``:
```
Comment (sigmavirus24) [inline docs/scenarios/json.rst]: Above we address the user (e.g., You), we should keep the same voice. Further, I'm not sure we need the examples below.

File: docs/scenarios/json.rst [hunk 10]
```diff
@@ -65,74 +70,22 @@ As well as write to them:
...
-        'first_name': 'Guido',
-        'second_name': 'Rossum'
-    }
-    print simplejson.dumps(d)
-    "{'first_name':'Guido','last_name':'Rossum'}"
-
+    pip install simplejson
 
-It is also possible to import JSON files:
+Usage
```
Comment (sigmavirus24) [inline docs/scenarios/json.rst]: Again, I really don't think this is necessary.

**Comment by royka** [comment]
@sigmavirus24 Thanks for reviewing looking at the comments now

File: docs/scenarios/json.rst (hunk 3 above)
Comment (royka) [inline docs/scenarios/json.rst]: yeah this bit I'm not too sure about either, the XML page doesn't have a description for XML. I think then most of the JSON description can be removed

File: docs/scenarios/json.rst (hunk 10 above)
Comment (royka) [inline docs/scenarios/json.rst]: ok

File: docs/scenarios/json.rst (hunk 9 above)
Comment (royka) [inline docs/scenarios/json.rst]: Do you mean the .loads and .dumps examples? I suppose they could be removed. What do we get by the examples being here, not alot I guess.  Maybe I could replace all the examples with a paragraph stating that JSON can be loaded and dumped to files with the library too.

Comment (royka) [inline docs/scenarios/json.rst]: but then I mention that JSON can be loaded and dumped from files at the start. So I think we can remove this section

**Comment by royka** [comment]
Updated based on your feedback @sigmavirus24 :)

**Comment by sigmavirus24** [comment]
Thanks @royka

**Comment by royka** [comment]
Thanks very much @sigmavirus24

**Comment by kennethreitz** [comment]
@sigmavirus24 should we merge this?

**Comment by sigmavirus24** [comment]
@kennethreitz it is merged already?

**Comment by royka** [comment]
@sigmavirus24  @kennethreitz  I think it has been merged

**Pull request merged**